solc/CommandLineInterface.cpp

```diff
@@ -760,8 +769,78 @@ bool CommandLineInterface::processInput()
 			m_compiler->useMetadataLiteralSources(true);
 		if (m_args.count(g_argInputFile))
 			m_compiler->setRemappings(m_args[g_argInputFile].as<vector<string>>());
-		for (auto const& sourceCode: m_sourceCodes)
-			m_compiler->addSource(sourceCode.first, sourceCode.second);
+
+		if (m_args.count(g_argImportAst))
+		{
+			//read file contents, which either are
+			// a) a json-file with multiple sources (as produced by --combined-json)
+			// b) a literal metadata-json-file with SolidityAST as language (as produced by --metadata --metadata-literal --import-ast)
+			// c) a json-file with one source only (as produced --ast-combined-json)
```

Comment:
I thought that it does not make sense to restrict the import to only work for one specific kind of JSON-file, moving it in a helper function does clean it up a lot though :)